### PR TITLE
runtime: config program fixes (pt. 1)

### PR DESCRIPTION
**Pull request by ripatel-fd**
- Add missing length restriction for instruction data ##fuzz
- Reject malformed bools ##fuzz
- Fix sequence of writable checks ##fuzz
- Config program should consume 450 CU ##fuzz

### CODE REVIEW

File: src/flamenco/runtime/program/fd_config_program.c [hunk 1]
```diff
@@ -25,7 +39,12 @@ _process_config_instr( fd_exec_instr_ctx_t ctx ) {
 
   fd_config_keys_t key_list;
   int decode_result = fd_config_keys_decode( &key_list, &decode );
-  if( decode_result != FD_BINCODE_SUCCESS )
+  if( decode_result != FD_BINCODE_SUCCESS ||
+      (ulong)ctx.instr->data + 1232UL < (ulong)decode.data )
+    return FD_EXECUTOR_INSTR_ERR_INVALID_INSTR_DATA;
```
Comment (topointon-jump) [inline src/flamenco/runtime/program/fd_config_program.c:44]: Do you mind documenting the reason for this and the magic number? Ty 🙏

File: src/flamenco/runtime/program/fd_config_program.c [hunk 2]
```diff
@@ -164,20 +185,33 @@ _process_config_instr( fd_exec_instr_ctx_t ctx ) {
...
-     https://github.com/solana-labs/solana/blob/v1.17.17/programs/config/src/config_processor.rs#L128-L133 */
+     https://github.com/solana-labs/solana/blob/v1.17.17/programs/config/src/config_processor.rs#L128-L129 */
 
   if( FD_UNLIKELY( !fd_borrowed_account_acquire_write( config_acc_rec ) ) )
     return FD_EXECUTOR_INSTR_ERR_ACC_BORROW_FAILED;
 
-  if( FD_UNLIKELY( !fd_instr_acc_is_writable_idx( ctx.instr, 0 ) ) )
-    return FD_EXECUTOR_INSTR_ERR_READONLY_DATA_MODIFIED;
+  /* Upgrade to writable handle
+     https://github.com/solana-labs/solana/blob/v1.17.17/programs/config/src/config_processor.rs#L130-L133 */
```
Comment (topointon-jump) [inline src/flamenco/runtime/program/fd_config_program.c:194]: 🙌

File: src/flamenco/runtime/program/fd_config_program.c [hunk 3]
```diff
@@ -164,20 +185,33 @@ _process_config_instr( fd_exec_instr_ctx_t ctx ) {
...
+  /* Inlined solana_sdk::transaction_context::BorrowedAccount::get_data_mut */
 
-  int modify_err = fd_instr_borrowed_account_modify_idx( &ctx, 0, config_acc_rec->const_meta->dlen, &config_acc_rec );
-  if( FD_UNLIKELY( modify_err!=FD_ACC_MGR_SUCCESS ) )
-    return FD_EXECUTOR_INSTR_ERR_READONLY_DATA_MODIFIED;
+  do {
+    int err;
+    if( !fd_account_can_data_be_changed( ctx.instr, 0, &err ) )
+      return err;
+  } while(0);
```
Comment (topointon-jump) [inline src/flamenco/runtime/program/fd_config_program.c:207]: Wondering if we should make these do..while loops a macro? So it's more clear why they are there

File: src/flamenco/runtime/program/fd_config_program.c [hunk 4]
```diff
@@ -192,6 +226,11 @@ _process_config_instr( fd_exec_instr_ctx_t ctx ) {
 
 int
 fd_config_program_execute( fd_exec_instr_ctx_t ctx ) {
+  do {
+    int err = fd_exec_consume_cus( ctx.txn_ctx, 450UL );
```
Comment (topointon-jump) [inline src/flamenco/runtime/program/fd_config_program.c:230]: nit: magic number

**Review by topointon-jump — APPROVED** [review]
Awesome! Feel free to ignore comments

File: src/flamenco/runtime/program/fd_config_program.c (hunk 4 above)
Comment (topointon-jump) [reply src/flamenco/runtime/program/fd_config_program.c:230]: Oh just define a constant with a link to Solana: https://github.com/solana-labs/solana/blob/2afde1b028ed4593da5b6c735729d8994c4bfac6/programs/config/src/config_processor.rs#L14

File: src/flamenco/runtime/program/fd_config_program.c (hunk 3 above)
Comment (topointon-jump) [reply src/flamenco/runtime/program/fd_config_program.c:207]: I was wondering if `ERR_SCOPE` or something would be clearer, but up to you. Don't feel strongly either way :)

**Review by topointon-jump — APPROVED** [review]

**Comment by ripatel-fd** [comment]
Thanks for reviewing @topointon-jump 🙏

**Pull request merged**